File: files/2026-01-25/CSEC 25012026.py
import pandas as pd
import numpy as np

path = "2026-01-25/Challenge 100.xlsx"
input_df = pd.read_excel(path, sheet_name=1, usecols="B", skiprows=2, nrows=3)
multi = pd.read_excel(path, sheet_name=1, usecols="C", skiprows=1, nrows=1, header=None).iloc[0, 0]
test = pd.read_excel(path, sheet_name=1, usecols="E", skiprows=1, nrows=10)

result = input_df.assign(Items=input_df['Items'].map(lambda x: list(np.repeat(x, multi)))).explode('Items').reset_index(drop=True)

print(result['Items'].equals(test['Solution'])) # True

File: files/2026-01-25/CSEC 25012026.R
library(tidyverse)
library(readxl)
library(unpivotr)

path <- "2026-01-25/Challenge 100.xlsx"
input <- read_excel(path, sheet = 2, range = "B3:B6")
multi <- read_excel(path, sheet = 2, range = "C2:C2", col_names = FALSE) %>%
  pull(1)
test <- read_excel(path, sheet = 2, range = "E2:E11")

result = input %>%
  mutate(Items = map(Items, ~ rep(.x, multi))) %>%
  unnest()

all.equal(result$Items, test$Solution)
# [1] TRUE
